### DISCUSSION TITLE
containerMode Kubernetes: new pods get ErrImagePull

### DISCUSSION THREAD

**Discussion by jmbravo** [actions/actions-runner-controller#3348] [Questions]
Hi,

I have changed the containerMode from dind to Kubernetes.

The problem is that when I launch the workflow the newly created pod cannot pull the image, why can this happen?

It seems to be a DNS error. On the other hand, the "parent" pod has configured the secret regcred and in dind mode it was pulling without problem.

What could have happened so that in kubernetes mode it can't pull?

Thanks!

  ----     ------   ----  ----     ------- Normal   Pulling  24s   kubelet  Pulling image "artifactory.mycompany.com/cloudops-images/ubuntu-sqlplus:1.0" Warning  Failed   24s   kubelet  Failed to pull image "artifactory.mycompany.com/cloudops-images/ubuntu-sqlplus:1.0": failed to pull and unpack image "artifactory.mycompany.com/cloudops-images/ubuntu-sqlplus:1.0": failed to resolve reference "artifactory.mycompany.com/cloudops-images/ubuntu-sqlplus:1.0": failed to do request: Head "https://artifactory.mycompany.com/v2/cloudops-images/ubuntu-sqlplus/manifests/1.0": dial tcp: lookup artifactory.mycompany.com on 10.77.252.41:53: no such host Warning  Failed   24s   kubelet  Error: ErrImagePull Normal   BackOff  24s   kubelet  Back-off pulling image "artifactory.mycompany.com/cloudops-images/ubuntu-sqlplus:1.0" Warning  Failed   24s   kubelet  Error: ImagePullBackOff

Edit: my container registry has public IP so I need nameserver 8.8.8.8, I don't know why it's trying 10.77.252.41:53

**Comment by rarecrumb** [comment]
> Hi,
> 
> I have changed the containerMode from dind to Kubernetes.

What caused you to make the switch away from DinD to Kubernetes? I'm curious if there was a performance increase.

> The problem is that when I launch the workflow the newly created pod cannot pull the image, why can this happen?
> 
> It seems to be a DNS error. On the other hand, the "parent" pod has configured the secret regcred and in dind mode it was pulling without problem.
> 
> What could have happened so that in kubernetes mode it can't pull?
> 
> Thanks!
> 

Yeah, this looks like a DNS resolution problem.

> Edit: my container registry has public IP so I need nameserver 8.8.8.8, I don't know why it's trying 10.77.252.41:53

Under your GHA runner `spec`, you might try using Pod's `dnsConfig`: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config to point to the DNS server you'd like to use.
